### PR TITLE
eimplement regression fix #55

### LINKED ISSUE #55: Known Issue: Regression
There is a known issue with regression. It occurs due to some type checking (some things require a jStat object while others don't etc). It will get fixed in future.

### PR DESCRIPTION
**Pull request by yiyuezhuo**
add core function slice
add core function diagonal to create diagonal matrix
add core function sliceAssign
add core function copy
add core function cola,rowa
enhance core function col,row to support list select

add helper function triaUpSolve,triaLowSolve to solve triangle matrix
equation
implement lu,cholesky decomposition
reimplement QR decomposition function
add linearalgebra function outer to do outer product operation
add linearalgebra function lstsq to support regress

add file models.js to reimplement OLS regress by QR decomposition.
Now regree provide t-test,f-test,95% confidence interval,R2,adjust R2.

add some test file for modified thing.

### CODE REVIEW

File: dist/jstat.js [hunk 1]
```diff
@@ -144,20 +144,49 @@ jStat.dimensions = function dimensions(arr) {
...
 jStat.row = function row(arr, index) {
+  if(isArray(index)){
+	  return index.map(function(i){
+		  return jStat.row(arr,i);
+	  })
+  }
   return arr[index];
 };
 
+jStat.rowa=function (arr,i){
```
Comment (trevnorris) [inline dist/jstat.js]: Mind adding a comment with a brief description about what it does or returns? Same with any other new functions that don't have one.

**Comment by trevnorris** [comment]
There's a lot of great stuff here. Mind adding entries to `doc/md` so others know it's available?

**Comment by yiyuezhuo** [comment]
add document

**Comment by trevnorris** [comment]
i'm working on getting this landed, but almost all of it isn't formatted correctly. Part of that is on me for not having a linter integrated. So attempting to fix the issues before landing, but there are quite a few so not sure when i'll have it finished.

**Comment by yiyuezhuo** [comment]
replace all tabs to two space

**Comment by trevnorris** [comment]
Style fixed, commits squashed and landed in 8d713bef860f336cf09cc01ca0a28b33dc38c3e0. Released in v1.5.3. Thanks for the code!

**Comment by yiyuezhuo** [comment]
Thanks for style fixed, when I wrote the code first I did't know the importance.

When I test regression on 1000+ dataset I found efficiency problem 
so I improve algorithm now it can run on larger dataset.

Another newbie problem,How I build document on Windows (to test whether new document chapter `regression` is exist.)? See issue #195 build document fail

**Comment by trevnorris** [comment]
@yiyuezhuo 
> Thanks for style fixed, when I wrote the code first I did't know the importance.

No worries. I should have a linter in with tests.

Seems I landed this before you were able to do the optimization. There are unfortunately too many conflicts to land this cleanly because of a divergent commit history:
```
*   19a713f (yiyuezhuo/master) fix format problem and accelerate QR decomposition
|\  
| * 44686cd (HEAD -> tag: 1.5.3, origin/master, origin/HEAD, master) Release 1.5.3
| * 8d713be Fix QR() implementation and other additions
* | 3018e0d replace tab and accelerate QR decomposition
* | 9c3ceba add document
* |   0b3c1e3 Merge remote-tracking branch 'upstream/master'
```
If you like, you can simply tell me what functions you optimized and I'll go ahead and patch just those changes in. Or you can reset your `master` to the current upstream's `master` and reapply the changes. I'll leave that up to you. Thanks much for paying attention to the performance.

**Comment by yiyuezhuo** [comment]
I have reset and reapply the change.Is it can work now?

**Comment by trevnorris** [comment]
Sorry for the delay. Thanks for that. I'll be able to apply the applicable changes.

**Comment by trevnorris** [comment]
Landed in 4c91a88be498e026d0f9274d9920d1a09224e205

**Comment by trevnorris** [comment]
Released in [v1.6.0](https://github.com/jstat/jstat/releases/tag/1.6.0)

**Pull request closed without merging**